tests/integration-tests/parsing/test_parsing_service.py

```
@@ -0,0 +1,431 @@
+            mock_project_service.update_project_status.assert_called_once()
+            call_args = mock_project_service.update_project_status.call_args
+            assert call_args[0][1] == ProjectStatusEnum.ERROR or str(call_args[0][1]).lower() == "error"
+
+
+class TestNeo4jFailures:
+    """Test Neo4j connection and operation failures (Part 4.4 of plan)."""
+
+    @pytest.mark.asyncio
+    async def test_neo4j_connection_failure_on_cleanup(self, db_session):
+        """Neo4j connection failure during cleanup_graph → status ERROR."""
+        from neo4j.exceptions import ServiceUnavailable
```

Ruff (0.15.4)

[error] Redefinition of unused `ServiceUnavailable` from line 11: `ServiceUnavailable` redefined here

Remove definition: `ServiceUnavailable`

(F811)